source/MaterialXRenderGlsl/GlslValidator.cpp

```diff
@@ -349,65 +345,40 @@ void GlslValidator::validateInputs()
 ////////////////////////////////////////////////////////////////////////////////////
 // Binders
 ////////////////////////////////////////////////////////////////////////////////////
-
-void GlslValidator::updateViewInformation()
-{
-    // Assume identify for model's world matrix
-    Matrix44& modelMatrix = _viewHandler->worldMatrix();
-    modelMatrix = Matrix44::IDENTITY;
-
-    Matrix44& viewMatrix = _viewHandler->viewMatrix();
-    viewMatrix = Matrix44::IDENTITY;
-
-    Vector3& viewDirection = _viewHandler->viewDirection();
-    viewDirection[0] = 0.0f;
-    viewDirection[1] = 0.0f;
-    viewDirection[2] = 1.0f;
-
-    Vector3& viewPosition = _viewHandler->viewPosition();
-
-    // Offset view position a little beyond geometry bounds
-    Vector3 minBounds = _geometryHandler->getMinimumBounds();
-    float distance = minBounds.getMagnitude() + 0.5f;
-
-    viewPosition[0] = 0.0f;
-    viewPosition[1] = 0.0f;
-    viewPosition[2] = -distance;
-
-    viewMatrix = viewMatrix.createTranslation(viewPosition);
-
-    // Update projection matrix
-    if (_orthographicView)
-    {
-        _viewHandler->setOrthoGraphicProjectionMatrix(-1.0f, 1.0f, -1.0f, 1.0f, NEAR_PLANE_ORTHO, FAR_PLANE_ORTHO);
-    }
-    else
-    {
-        float aspectRatio = (float)_frameBufferWidth / (float)_frameBufferHeight;
-        _viewHandler->setPerspectiveProjectionMatrix(FOV_PERSP, aspectRatio, NEAR_PLANE_PERSP, FAR_PLANE_PERSP);
-    }
-}
-
-void GlslValidator::bindFixedFunctionViewInformation()
+void GlslValidator::updateViewInformation(const Vector3& eye,
+                                          const Vector3& center,
+                                          const Vector3& up,                                          
+                                          float viewAngle,
+                                          float nearDist,
+                                          float farDist,
+                                          float objectScale)
 {
-    // Bind projection
-    glMatrixMode(GL_PROJECTION);
-    Matrix44& projection = _viewHandler->projectionMatrix();
-    glLoadMatrixf(&(projection[0][0]));
-
-    // Bind model view matrix
-    glMatrixMode(GL_MODELVIEW);
-    Matrix44 viewMatrix = _viewHandler->viewMatrix();
-    // Note: (model * view) is just Identity * view.
-    glLoadMatrixf(&(viewMatrix[0][0]));
-
-    checkErrors();
+    const float PI = std::acos(-1.0f);
```

Comment:
Is this necessary? Doesn't math.h define M_PI?

Reply:
I got this from jstone, so assuming this is what he prefers. Better precision I guess?

Reply:
Ok